vrf: Ignore MAC and accept_all_mac_addresses=false

cathay4t opened the pull request:
The VRF does not support changing MAC address and
accept_all_mac_addresses mode.

This patch ignore mac address in desire state with warning.
This patch also ignore `ACCEPT_ALL_MAC_ADDRESSES=False` in desire state
without any warning.

For `ACCEPT_ALL_MAC_ADDRESSES=True`, user will get verification error
which unrelated to this patch.

Unit and integration test cases are included.

ffmancera commented on libnmstate/ifaces/vrf.py:
```diff
@@ -46,8 +49,10 @@ def is_virtual(self):
     def pre_edit_validation_and_cleanup(self):
         super().pre_edit_validation_and_cleanup()
-        if self.is_up:
+        if self.is_up and self.is_desired:
```
Should we validate and cleanup if it is only changed but not desired? What fo you think?

cathay4t replied:
Changed to `if self.is_up and (self.is_desired or self.is_changed):`

ffmancera replied:
Thanks!

cathay4t committed:
vrf: Ignore MAC and accept_all_mac_addresses=false

The VRF does not support changing MAC address and
accept_all_mac_addresses mode.

This patch ignore mac address in desire state with warning.
This patch also ignore `ACCEPT_ALL_MAC_ADDRESSES=False` in desire state
without any warning.

For `ACCEPT_ALL_MAC_ADDRESSES=True`, user will get verification error
which unrelated to this patch.

Unit and integration test cases are included.

Signed-off-by: Gris Ge <fge@redhat.com>

ffmancera reviewed (approved):
LGTM, thanks!
